app/assets/javascripts/discourse/app/components/search-menu.js

```diff
@@ -0,0 +1,302 @@
+      //this.sendWidgetEvent("linkClicked");
+      DiscourseURL.routeTo(url);
+    }
+  }
+
+  //@action
+  //updateInTopicContext(value) {
+  //this.inTopicContext = value;
+  //}
+
+  @action
+  updateTypeFilter(value) {
+    this.typeFilter = value;
+  }
+
+  @action
+  clearPMInboxContext() {
+    this.inPMInboxContext = false;
+  }
+
+  @action
+  setTopicContext() {
+    this.inTopicContext = true;
+  }
+
+  @action
+  clearTopicContext() {
+    this.inTopicContext = false;
+  }
+
+  // for cancelling debounced search
+  cancel() {
+    if (this._activeSearch) {
+      this._activeSearch.abort();
+      this._activeSearch = null;
+    }
+  }
+
+  perform() {
+    this.cancel();
+
+    const matchSuggestions = this.matchesSuggestions();
+    if (matchSuggestions) {
+      this.noResults = true;
+      this.results = {};
+      this.loading = false;
+      this.suggestionResults = [];
+
+      if (matchSuggestions.type === "category") {
+        const categorySearchTerm = matchSuggestions.categoriesMatch[0].replace(
+          "#",
+          ""
+        );
+
+        const categoryTagSearch = searchCategoryTag(
+          categorySearchTerm,
+          this.siteSettings
+        );
+        Promise.resolve(categoryTagSearch).then((results) => {
+          if (results !== CANCELLED_STATUS) {
+            this.suggestionResults = results;
+            this.suggestionKeyword = "#";
+          }
+        });
+      } else if (matchSuggestions.type === "username") {
+        const userSearchTerm = matchSuggestions.usernamesMatch[0].replace(
+          "@",
+          ""
+        );
+        const opts = { includeGroups: true, limit: 6 };
+        if (userSearchTerm.length > 0) {
+          opts.term = userSearchTerm;
+        } else {
+          opts.lastSeenUsers = true;
+        }
+
+        userSearch(opts).then((result) => {
+          if (result?.users?.length > 0) {
+            this.suggestionResults = result.users;
+            this.suggestionKeyword = "@";
+          } else {
+            this.noResults = true;
+            this.suggestionKeyword = false;
+          }
+        });
+      } else {
+        this.suggestionKeyword = matchSuggestions[0];
+      }
+      return;
+    }
+
+    this.suggestionKeyword = false;
+
+    if (!this.search.activeGlobalSearchTerm) {
+      this.noResults = false;
+      this.results = {};
+      this.loading = false;
+      this.invalidTerm = false;
+    } else if (
+      !isValidSearchTerm(this.search.activeGlobalSearchTerm, this.siteSettings)
+    ) {
+      this.noResults = true;
+      this.results = {};
+      this.loading = false;
+      this.invalidTerm = true;
+    } else {
+      this.invalidTerm = false;
+
+      this._activeSearch = searchForTerm(this.search.activeGlobalSearchTerm, {
+        typeFilter: this.typeFilter,
+        fullSearchUrl: this.fullSearchUrl,
+        searchContext: this.searchContext,
+      });
+
+      this._activeSearch
+        .then((results) => {
```

Comment:
This would require us to update 

https://github.com/discourse/discourse/blob/7958f571748cd405fb750a4b871d6f0d8c1f92a2/app/assets/javascripts/discourse/app/lib/search.js#L143

which would break the widget implementation... I think we should leave this as is.